### ISSUE TITLE
Upgrading to 1.0.0 causes server hangup

### ISSUE THREAD

**Issue by divanikus** [fluent/fluent-bit#972]
## Bug Report

**Describe the bug**
Upgrading from 0.13.8 to 1.0.0 caused 100% cpu usage, whole memory consumption and unresponsiveness of the server.

**To Reproduce**
- Upgrade to 1.0.0
```
[499726.235043] Out of memory: Kill process 29664 (td-agent-bit) score 817 or sacrifice child
[499726.235107] Killed process 29664 (td-agent-bit) total-vm:131258644kB, anon-rss:111353612kB, file-rss:0kB, shmem-rss:4kB
[499729.740269] oom_reaper: reaped process 29664 (td-agent-bit), now anon-rss:0kB, file-rss:4kB, shmem-rss:0kB
```

**Your Environment**
* Version used: 1.0.0
* Server type and version: Hardware
* Operating System and version: Debian 9
* Filters and plugins: in_systemd, in_tail, in_kmsg

```
[SERVICE]
    Flush        5
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf
    Parsers_File parsers_extra.conf
    Plugins_File plugins.conf

    HTTP_Server  Off
    HTTP_Listen  0.0.0.0
    HTTP_Port    2020

[INPUT]
    Name   kmsg
    Tag    kmsg

[INPUT]
    Name   systemd
    Tag    systemd
    DB     /var/tmp/journal.db
    DB.Sync Off

[INPUT]
    Name   tail
    Tag    fail2ban
    Parser fail2ban
    Path   /var/log/fail2ban.log
    DB     /var/tmp/journal.db
    DB.Sync Off

[FILTER]
    Name   modify
    Match  systemd
    Rename MESSAGE message
    Rename PRIORITY priority

[FILTER]
    Name   record_modifier
    Match  *
    Record hostname master0

[OUTPUT]
    Name   forward
    Match  *
    Host   td-agent.example.com
    Port   24224
```
```
[PARSER]
    Name   apache
    Format regex
    Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
    Time_Key time
    Time_Format %d/%b/%Y:%H:%M:%S %z

[PARSER]
    Name   apache2
    Format regex
    Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>.*)")?$
    Time_Key time
    Time_Format %d/%b/%Y:%H:%M:%S %z

[PARSER]
    Name   apache_error
    Format regex
    Regex  ^\[[^ ]* (?<time>[^\]]*)\] \[(?<level>[^\]]*)\](?: \[pid (?<pid>[^\]]*)\])?( \[client (?<client>[^\]]*)\])? (?<message>.*)$

[PARSER]
    Name   nginx
    Format regex
    Regex ^(?<remote>[^ ]*) (?<host>[^ ]*) (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
    Time_Key time
    Time_Format %d/%b/%Y:%H:%M:%S %z

[PARSER]
    Name   json
    Format json
    Time_Key time
    Time_Format %d/%b/%Y:%H:%M:%S %z

[PARSER]
    Name         docker
    Format       json
    Time_Key     time
    Time_Format  %Y-%m-%dT%H:%M:%S.%L
    Time_Keep    On
    # Command      |  Decoder | Field | Optional Action
    # =============|==================|=================
    Decode_Field_As   escaped    log

[PARSER]
    Name        docker-daemon
    Format      regex
    Regex       time="(?<time>[^ ]*)" level=(?<level>[^ ]*) msg="(?<msg>[^ ].*)"
    Time_Key    time
    Time_Format %Y-%m-%dT%H:%M:%S.%L
    Time_Keep   On

[PARSER]
    Name        syslog-rfc5424
    Format      regex
    Regex       ^\<(?<pri>[0-9]{1,5})\>1 (?<time>[^ ]+) (?<host>[^ ]+) (?<ident>[^ ]+) (?<pid>[-0-9]+) (?<msgid>[^ ]+) (?<extradata>(\[(.*)\]|-)) (?<message>.+)$
    Time_Key    time
    Time_Format %Y-%m-%dT%H:%M:%S.%L
    Time_Keep   On

[PARSER]
    Name        syslog-rfc3164-local
    Format      regex
    Regex       ^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$
    Time_Key    time
    Time_Format %b %d %H:%M:%S
    Time_Keep   On

[PARSER]
    Name        syslog-rfc3164
    Format      regex
    Regex       /^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$/
    Time_Key    time
    Time_Format %b %d %H:%M:%S
    Time_Format %Y-%m-%dT%H:%M:%S.%L
    Time_Keep   On

[PARSER]
    Name    mongodb
    Format  regex
    Regex   ^(?<time>[^ ]*)\s+(?<severity>\w)\s+(?<component>[^ ]+)\s+\[(?<context>[^\]]+)]\s+(?<message>.*?) *(?<ms>(\d+))?(:?ms)?$
    Time_Format %Y-%m-%dT%H:%M:%S.%L
    Time_Keep   On
    Time_Key time

[PARSER]
    # http://rubular.com/r/izM6olvshn
    Name crio
    Format Regex
    Regex /^(?<time>.+)\b(?<stream>stdout|stderr)\b(?<log>.*)$/
    Time_Key    time
    Time_Format %Y-%m-%dT%H:%M:%S.%N%:z
    Time_Keep   On

[PARSER]
    Name    kube-custom
    Format  regex
    Regex   var\.log\.containers\.(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$

[PARSER]
    Name    filter-kube-test
    Format  regex
    Regex   .*kubernetes.(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$

[PARSER]
    Name   fail2ban
    Format regex
    Regex  ^(?<time>\S* \S*)\s+(?<criteria>\S*)\s+\[(?<pid>[0-9]+)\]:\s+(?<priority>\S+)\s+\[(?<service>\S+)\]\s+(?<message>.*?)$
    Time_Key time
    Time_Format %Y-%m-%d %H:%M:%S.%L
    Time_Offset +0300
```

**Additional context**
The issue caused a full blown disaster to my cluster, when 50% of servers just stopped to respond without any reason. Reverting back to 0.14.9 fixed the problem.

**Comment by edsiper** [comment]
Do you see similar behavior with v1.0.1?

**Comment by divanikus** [comment]
I've updated several servers so far. Looks like 1.0.1 works better. At least it didn't caused the hangup.

**Comment by edsiper** [comment]
Considering this issue fixed with v1.0.1
